### PR TITLE
Use Renderer to generate the ConfigMap

### PR DESCRIPTION
**Pull request by slaperche-scality**
**Component**:

buildchain

**Context**: 

We want to generate SLS from the buildchain, and we have to shove huge string inside it while trying to keep it readable AND have a valid YAML in the end.

**Summary**:

Instead of creating/formatting the SLS content by hand, let's use a YAML renderer.

**Acceptance criteria**: 

You can build a working ISO

---

Closes: #2070

### CODE REVIEW

**Review by NicolasT — DISMISSED** [review]
LGTM. One minor concern: the SLS renderer takes a `content` object, whilst we often put multiple objects in a single SLS file as a YAML document stream (objects separated by `---`). This renderer could be more generalized by taking a list of objects as `objects` and render them accordingly.

**Comment by slaperche-scality** [comment]
> LGTM. One minor concern: the SLS renderer takes a `content` object, whilst we often put multiple objects in a single SLS file as a YAML document stream (objects separated by `---`). This renderer could be more generalized by taking a list of objects as `objects` and render them accordingly.

Good point, and shouldn't be too hard to support: I can add a new commit for that.

**Comment by slaperche-scality** [comment]
Done, I've added support for list of objects.

**Review by NicolasT — APPROVED** [review]
Not 100% sure I'm 100% comfortable with this approach, since now we have an SLS in the Salt tree that's not represented by any file in the 'source' `salt/` tree, but OK.

File: buildchain/buildchain/targets/serialize.py [hunk 1]
```diff
@@ -95,3 +121,64 @@ def _render(self) -> Callable[[Any, Path], None]:
...
+    dumper.add_representer( # type: ignore
+        YAMLDocument.Literal, _literal_representer
+    )
+    dumper.add_representer( # type: ignore
+        YAMLDocument.ByteString, _bytestring_representer
+    )
+    try:
+        dumper.open() # type: ignore
+        for document in data:
+            dumper.represent(document) # type: ignore
```
Comment (NicolasT) [inline buildchain/buildchain/targets/serialize.py:178]: I assume this then adds the `---`?

Comment (slaperche-scality) [reply buildchain/buildchain/targets/serialize.py:178]: Yes, I've tested locally.

And the code is highly inspired from [official PyYAML code](https://github.com/yaml/pyyaml/blob/master/lib/yaml/__init__.py#L278-L284).

**Pull request merged**